src/System.Linq/tests/OrderByTests.cs

```diff
@@ -165,28 +157,33 @@ public void SameKeysVerifySortStable()
         [Fact]
         public void OrderedToArray()
         {
-            Record[] source = new Record[]
+            var source = new []
             {
-                new Record{ Name = "Tim", Score = 90 },
-                new Record{ Name = "Robert", Score = 90 },
-                new Record{ Name = "Prakash", Score = 90 },
-                new Record{ Name = "Jim", Score = 90 },
-                new Record{ Name = "John", Score = 90 },
-                new Record{ Name = "Albert", Score = 90 },
+                new { Name = "Tim", Score = 90 },
+                new { Name = "Robert", Score = 90 },
+                new { Name = "Prakash", Score = 90 },
+                new { Name = "Jim", Score = 90 },
+                new { Name = "John", Score = 90 },
+                new { Name = "Albert", Score = 90 },
             };
-            Record[] expected = new Record[]
+            var expected = new []
             {
-                new Record{ Name = "Tim", Score = 90 },
-                new Record{ Name = "Robert", Score = 90 },
-                new Record{ Name = "Prakash", Score = 90 },
-                new Record{ Name = "Jim", Score = 90 },
-                new Record{ Name = "John", Score = 90 },
-                new Record{ Name = "Albert", Score = 90 },
+                new { Name = "Tim", Score = 90 },
+                new { Name = "Robert", Score = 90 },
+                new { Name = "Prakash", Score = 90 },
+                new { Name = "Jim", Score = 90 },
+                new { Name = "John", Score = 90 },
+                new { Name = "Albert", Score = 90 },
             };
 
             Assert.Equal(expected, source.OrderBy(e => e.Score).ToArray());
         }
 
+        [Fact]
+        public void EmptyOrderedToArray()
+        {
+            Assert.Empty(Enumerable.Empty<int>().OrderBy(e => e).ToArray());
+        }
         //FIXME: This will hang with a larger source. Do we want to deal with that case?
```

Comment:
This originates from a test in the legacy tests, but as said, it doesn't actually work with larger inputs. I'm not sure the test should be there at all (we should just hang on such bad comparers and consider a complaint about it PEBKAC) but I remained faithful to the legacy tests in leaving these two in.